backend/src/main/java/com/google/blackswan/mock/SimpleRelatedDataGenerator.java

```diff
@@ -0,0 +1,119 @@
+
+  private static final SimpleRelatedDataGenerator INSTANCE = new SimpleRelatedDataGenerator();
+
+  private Multimap<String, DataInfo> relatedDataMap;
+  private Map<String, Map<Timestamp, Integer>> csvDataCache;
+
+  private SimpleRelatedDataGenerator() {
+    relatedDataMap = ArrayListMultimap.create();
+    csvDataCache = new HashMap<>();
+    prefillRelatedData(); // TODO: Remove when connected to datastore.
+  }
+
+  /** 
+   * Simulates action of querying configs from datastore and adding to 
+   * relatedDataMap. Currently, manually adds these relationship.
+   */
+  private void prefillRelatedData() {
+    // TODO: Replace with call for querying configs from datastore.
+    relatedDataMap.put("Ramen", new DataInfo(RELATED_METRIC, "Udon", CONFIG_USERNAME));
+    relatedDataMap.put("Ramen", new DataInfo(RELATED_METRIC, "Pho", CONFIG_USERNAME));
+    // Multimap looks like this right now: {{Ramen,{Udon, Pho}}, ...}.
+  }
+
+  public List<RelatedData> getRelatedData(String metricName, String dimensionName, 
+      Timestamp startTime, Timestamp endTime) {
+    List<RelatedData> relatedDataList = new ArrayList<RelatedData>();
+    // TODO: Currently only tries to find related data for dimension, ex. Ramen -> Pho, 
+    //       does not include logic for metric, should be similar.
+    if (relatedDataMap.containsKey(dimensionName)) {
+      for (DataInfo relatedTopic : relatedDataMap.get(dimensionName)) {
+        Map<Timestamp, Integer> topicDataPoints = 
+            getTopicDataPoints(relatedTopic.getDimensionName());
+
+        Map<Timestamp, MetricValue> dataPointsPlot = 
+            getDataPointsInRange(topicDataPoints, startTime, endTime);
+        if (dataPointsPlot.isEmpty()) {
+          continue;
+        }
+
+        relatedDataList.add(new RelatedData(
+            relatedTopic.getUsername(), relatedTopic.getMetricName(), 
+            relatedTopic.getDimensionName(), dataPointsPlot)
+          );        
+      }
+    }
+    return relatedDataList;
+  }
+
+  private ImmutableMap<Timestamp, Integer> getTopicDataPoints(String topic) {
+    if (!csvDataCache.containsKey(topic)) {
+      csvDataCache.put(topic, CSVParser.parseCSV(
+        SimpleRelatedDataGenerator.class.getResourceAsStream(FILE_LOCATION.get(topic))
+      ));
+    }
+
+    return ImmutableMap.copyOf(csvDataCache.get(topic));
+  }
+
+  private static ImmutableMap<Timestamp, MetricValue> getDataPointsInRange
+      (Map<Timestamp, Integer> topicDataPoints, Timestamp startTime, Timestamp endTime) {
+    
+    List<Timestamp> listKeys = new ArrayList<Timestamp>(topicDataPoints.keySet());
+    int indexOfStart = listKeys.indexOf(startTime);
+    int indexOfEnd = listKeys.indexOf(endTime);
+
+    if (indexOfStart == -1 && indexOfEnd == -1) {
+      return ImmutableMap.of();
+    }
+    
+    indexOfStart = indexOfStart == -1 ? 0 : indexOfStart;
+    indexOfEnd = indexOfEnd == -1 ? listKeys.size() - 1 : indexOfEnd;
```

Comment:
to confirm - these are when there might not be data before the anomaly's time range, or there might not be data after the anomaly's time range?

Reply:
yes, that is correct!